File: Quivi.Backoffice.Api/Dtos/ConfigurableField.cs
using Quivi.Domain.Entities;

namespace Quivi.Backoffice.Api.Dtos
{
    public class ConfigurableField
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public bool IsRequired { get; init; }
        public bool IsAutoFill { get; init; }
        public PrintedOn PrintedOn { get; init; }
        public AssignedOn AssignedOn { get; init; }
        public string? DefaultValue { get; init; }
        public ConfigurableFieldType Type { get; init; }
        public required IReadOnlyDictionary<Language, ConfigurableFieldTranslation> Translations { get; init; }
    }

    public class ConfigurableFieldTranslation
    {
        public required string Name { get; init; }
    }

    public enum ConfigurableFieldType
    {
        Text = 0,
        LongText = 1,
        Check = 2,
        Number = 3,
    }

    [Flags]
    public enum PrintedOn
    {
        None = 0,
        PreparationRequest = 1 << 1,
        TableBill = 1 << 2,
    }

    [Flags]
    public enum AssignedOn
    {
        None = 0,
        PoSSessions = 1 << 1,
        Ordering = 1 << 2,
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/configurableFields/ConfigurableField.ts
import { Language } from "../Language";
import { AssignedOn } from "./AssignedOn";
import { ConfigurableFieldType } from "./ConfigurableFieldType";
import { PrintedOn } from "./PrintedOn";

export interface ConfigurableField {
    readonly id: string;
    readonly name: string;
    readonly isRequired: boolean;
    readonly isAutoFill: boolean;
    readonly defaultValue?: string;
    readonly printedOn: PrintedOn;
    readonly assignedOn: AssignedOn;
    readonly type: ConfigurableFieldType;
    readonly translations: Record<Language, ConfigurableFieldTranslation>;
}

export interface ConfigurableFieldTranslation {
    readonly name: string;
}